src/compas_fab/robots/configuration.py

```diff
@@ -84,6 +148,41 @@ def __repr__(self):
         """Printable representation of :class:`Configuration`."""
         return self.__str__()
 
+    def __getitem__(self, item):
+        for name, value in zip(self.joint_names, self.joint_values):
+            if name == item:
+                return value
+        raise KeyError(item)
+
```

Comment:
Please remind me why aren't we adding a `__setitem__`? I think we talked about it, but now it feels like it's totally missing and I cannot recall why it would be problematic (maybe related to optional joint names, but in that case, getter accessor doesn't work either). I tested this and superficially seems to work:

```suggestion

    def __setitem__(self, item, value):
        i = self.joint_names.index(item)
        if i < 0:
            raise KeyError(item)
        self.joint_values[i] = value

```

Reply:
I think the reason had to do with user experience and expectations and being consistent with being a "read-only" dictionary rather than the worry of things breaking.  If this is added, then you can write things to the dictionary, but only sometimes.  Without knowing the internals, maybe it would be confusing to a user which aspects of a dictionary does a configuration have and which doesn't it.  "read-only" is succinct, but limiting.  "readable, and also over-writable, but not generally writable" is a bit much to explain.

Reply:
...but currently is not read-only either, it's only fixed-length but values can be changed (otherwise we could use tuples instead of `FixedLengthList`)

Reply:
yes, true, but... the writability is not a function of its dictionary-like qualities.  in any case, i don't mind adding the `__setitem__`.